msteams-platform/toolkit/TeamsFx-CLI.md

```diff
@@ -1,23 +1,462 @@
+
+* **Teams Toolkit CLI**: Accelerates Teams application development. It also enables CI/CD scenario where you can integrate CLI in scripts for automation.
+
+## Get started
+
+Install `teamsapp` from `npm` and run `teamsapp -h` to check all available commands:
+
+```bash
+  npm install -g @microsoft/teamsapp-cli
+  teamsapp -h
+```
+
+## Supported commands
+
+| Command | Description |
+|----------------|-------------|
+| `teamsapp doctor` | Prerequisites to create Teams application.|
+| `teamsapp new`| Create new Teams application.|
+| `teamsapp add`| Adds features to your Teams application.|
+| `teamsapp auth`| Manage cloud service accounts. The supported cloud services are Azure and Microsoft 365. |
+| `teamsapp entra-app` | Manage the Microsoft Entra app in the current application. |
+| `teamsapp env` | Manage environments. |
+| `teamsapp help` |  Show Microsoft Teams Toolkit CLI help. |
+| `teamsapp install` | Upload a custom app for a given application package across Microsoft 365. |
+| `teamsapp launchinfo` | Get launch information of an acquired Microsoft 365 App. |
+| `teamsapp list` | List available Microsoft Teams application templates and samples. |
+| `teamsapp provision` | Provision cloud resources in the current application.|
+| `teamsapp deploy` | Deploy the current application. |
+| `teamsapp package` | Build your Teams app into package for publishing.|
+| `teamsapp validate` | Validate the current application.|
+| `teamsapp publish` | Publish the app to Teams.|
+| `teamsapp preview` | Preview the current application. |
+| `teamsapp update` |  Update the Teams app manifest to Teams Developer Portal. |
+| `teamsapp upgrade` | Upgrade the project to work with the latest version of Teams Toolkit. |
+| `teamsapp collaborator`| Collaborate with other developers in same project.|
+| `teamsapp uninstall` | Remove an acquired Microsoft 365 App. |
+
+## Interactive mode
+
+`teamsapp` command runs in interactive mode by default. You can work in non-interactive mode by setting `--interactive` flag to `false`.
+
+## `teamsapp doctor`
+
+Prerequiste checker for building Microsoft Teams apps.
+
+## `teamsapp new`
+
+By default, `teamsapp new` is in interactive mode and guides to create new Teams application. You can work in non-interactive mode by setting `--interactive` flag to `false`.
+
+| Command | Description |
+|:----------------  |:-------------|
+| `teamsapp new sample` | Create an app from an existing sample. |
+| `teamsapp list samples` | List all the available samples. |
+
+### Parameters for `teamsapp new -h`
+
+| Parameter | Required | Description |
+|:---------------- |:-------------|:-------------|
+|`--app-name -n` | Yes| Name of your Teams application.|
+|`--interactive`| No | Select the options interactively. The available options are `true` and `false` and the default value is `true`.|
```

Comment:
We should separate options and global options. Global options are applicable for all commands, can be captured at top of this document, no need to document for each of the commands.

![image](https://github.com/MicrosoftDocs/msteams-docs/assets/11220663/22d71fef-8f73-42e5-9608-2c3d319f8dc1)

Reply:
Updated.